tests/scripts/check_test_cases.py

```diff
@@ -111,6 +112,17 @@ def walk_ssl_opt_sh(self, file_name):
                 self.process_test_case(descriptions,
                                        file_name, line_number, description)
 
+    def walk_compat_sh(self, file_name):
+        """Iterate over the test cases compat.sh with a similar format."""
+        descriptions = self.new_per_file_state() # pylint: disable=assignment-from-none
+        compat_cmd = ['sh', file_name, '--list-test-case']
+        compat_output = subprocess.check_output(compat_cmd)
+        # Assume compat.sh is responsible for printing identical format of
+        # test case description between --list-test-case and its OUTCOME.CSV
+        description = compat_output.strip().split(b'\n')
+        for idx, descrip in enumerate(description):
+            self.process_test_case(descriptions, file_name, idx, descrip)
```

Comment:
Just a minor comment:
idx here does not mean the line number here.

Reply:
I would personally prefer passing None than passing something incorrect, but a comment can do as well :)